jwst/mrs_imatch/mrs_imatch_step.py

```diff
@@ -337,17 +340,22 @@ def _match_models(models, channel, degree, center=None, center_cs='image'):
     # if requested:
     ##### model.meta.instrument.channel
 
-    # set 2D models' background meta info:
-    for im, poly in zip(models, bkg_poly_coef):
-        im.meta.background.subtracted = False
-        im.meta.background.polynomial_info.append(
-            {
+    if np.isnan(bkg_poly_coef).any():
+        bkg_poly_coef = None
+        for im in models:
+            im.meta.cal_step.mrs_imatch = 'SKIPPED'
```

Comment:
I wonder if it would be good to also set `im.meta.background.subtracted = False` here (as is done below), just in case anything downstream uses that value for anything (just to make sure it's set to something, rather than None). Or could the presence of a value (either True or False) trick something into thinking that the background has at least been fitted and in this case it hasn't?

Reply:
Maybe the code (downstream) should _be made/forced_ to check `im.meta.cal_step.mrs_imatch == 'COMPLETE'` before using `im.meta.background` values.